tools/src/shellcheck.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+SHELLCHECK_VERSION=v0.8.0
+
+GOARCH=$(go env GOARCH)
+GOOS=$(go env GOOS)
+
+if [[ "$GOARCH" == "amd64" || "$GOOS" == "darwin" ]]; then
```

Comment:
why is the `darwin` check needed ? I think above `if` will be true for M1 Macs (`darwin/arm64`) which we dont want

Reply:
In shell release page, there is only x64 binary for darwin

Reply:
shellcheck is supported for `darwin/arm64` via homebrew https://github.com/koalaman/shellcheck/issues/2229#issuecomment-858753596
if we merge this, we'll drop support for `darwin/arm64`, since the M1 Mac user cannot use their local `shellcheck` neither can they download it

Reply:
I don't think I like having a wrapper shellscript do tis.  `tools.mk` should just know how to download shellcheck and install that binary to `tools/bin/shellcheck`.

Here's Telepresence's rule for doing that: https://github.com/telepresenceio/telepresence/blob/4c6fc8953adc9012a3a5b85856a9802c51e84cac/build-aux/tools.mk#L80-L97

Reply:
IIRC, despite "x86_64" being in the filename, the `darwin.x86_64` file is actually a "fat" binary with compiled code for both x86_64 and arm64.

Reply:
while @Xunzhuo addresses @LukeShu 's suggestions, @chauhanshubham can you help verify these PR changes on a `darwin/arm64` machine ?

Reply:
I am using M1 Mac, it works fine.